src/main/java/net/snowflake/ingest/utils/HttpUtil.java

```diff
@@ -29,7 +35,8 @@ static public HttpClient getHttpClient()
 
     return httpClient;
```

Comment:
I know you did not write this code, but getHttpClient() method should be synchronized since this is static method and httpClient object is shared by multiple thread potentially. This is by design since httpClient is thread-safe and we avoid httpClient per SimpleIngestManager. So if two threads are initializing SimpleIngestManager at the same time, there is a race condition here. You don't need to make the change immediately, but just keep in mind, there is a potential improvement here.